.github/workflows/java-gradle-release.yaml

```diff
@@ -0,0 +1,130 @@
+name: Reusable workflow for releasing java gradle projects
+
+on:
+  workflow_call:
+    inputs:
+      release-type:
+        description: "Scope of the release (major, minor or patch)."
+        required: true
+        type: string
+      java-distribution:
+        description: "Java distribution to be installed. (Default is temurin)"
+        required: false
+        type: string
+        default: "temurin"
+      java-version:
+        description: "Java version to be installed. (Default is 11)"
+        required: false
+        type: string
+        default: "11"
+      gradle-version:
+        description: "Gradle version to be installed. (Default is 6.7.1)"
+        required: false
+        type: string
+        default: "6.7.1"
+      gradle-cache:
+        description: "Gradle cache to use. Options are gradle, maven, sbt. (Default is gradle)"
+        required: false
+        type: string
+        default: "gradle"
+      working-directory:
+        description: "Working directory of your Gradle artifacts. (Default is root directory)"
+        required: false
+        type: string
+        default: "./"
+    secrets:
+      github-username:
+        description: "GitHub username for committing the changes."
+        required: true
+      github-email:
+        description: "GitHub email for committing the changes."
+        required: true
+      github-token:
+        description: "GitHub token for committing the changes."
+        required: true
+
+    # Map the workflow outputs to job outputs
+    outputs:
+      release-tag:
+        description: "The release tag."
+        value: ${{ jobs.evaluate-version.outputs.release-version }}
+
+jobs:
+  evaluate-version:
```

Comment:
Why is `evaluate-version` a separate job? To me, it seems like unnecessary overhead